app/frontend/javascript/survey_widget.js

```diff
@@ -0,0 +1,88 @@
+// This is a fairly brute-force
+// attempt to make sure that if
+// you don't want to fill in the survey
+// or already did fill it in, you
+// won't be asked again for a long time.
+
+// Despite our best efforts,
+// the survey was not registering
+// the fact that users had dismissed it
+// or already filled it in.
+
+// Note that SurveySparrow has its own
+// mechanisms in place to attempt to ensure
+// that users only fill out the survey once;
+// we just don't trust them.
+jQuery( document ).ready(function() {
+  if (jQuery('#ss_survey_widget').length != 1) {
+    return;
+  }
+  if (surveyAlreadyShown()) {
+    window.console.log("You've already clicked on the survey! Not gonna show it.")
+  }
+  else {
+    window.console.log("Launching the survey.")
+    sparrowLaunch({
+      //add custom params here
+    })
+  }
+  // Any click on the main survey div is
+  // interpreted as "user has started survey."
+  // Once they start the survey, never show
+  // this user the survey again for an entire year.
+  jQuery ('#ss_survey_widget').click(userHasStartedSurvey);
+});
+
+
+function surveyAlreadyShown() {
+  return document.cookie.match(/surveyAlreadyShown=true/) != null;
```

Comment:
Is this really the official way to check cookies from JS?  The `match` seems weird to me. Googling... maybe it is? Very weird.

Reply:
It's what we use for the "accept cookies" footer, and it does work. I agree it's very counterintuitive.